src/main/kotlin/com/terraformation/backend/accelerator/db/CohortStore.kt

```diff
@@ -0,0 +1,122 @@
+package com.terraformation.backend.accelerator.db
+
+import com.terraformation.backend.accelerator.model.CohortModel
+import com.terraformation.backend.accelerator.model.ExistingCohortModel
+import com.terraformation.backend.accelerator.model.NewCohortModel
+import com.terraformation.backend.accelerator.model.toModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.asNonNullable
+import com.terraformation.backend.db.default_schema.CohortId
+import com.terraformation.backend.db.default_schema.tables.daos.CohortsDao
+import com.terraformation.backend.db.default_schema.tables.pojos.CohortsRow
+import com.terraformation.backend.db.default_schema.tables.references.COHORTS
+import com.terraformation.backend.db.default_schema.tables.references.PARTICIPANTS
+import jakarta.inject.Named
+import java.time.InstantSource
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+import org.springframework.dao.DataIntegrityViolationException
+
+@Named
+class CohortStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val cohortsDao: CohortsDao,
+) {
+  fun fetchOneById(cohortId: CohortId): ExistingCohortModel {
+    return fetch(COHORTS.ID.eq(cohortId)).firstOrNull() ?: throw CohortNotFoundException(cohortId)
+  }
+
+  fun findAll(): List<ExistingCohortModel> {
+    return fetch(null)
+  }
+
+  fun create(model: NewCohortModel): ExistingCohortModel {
+    requirePermissions { createCohort() }
+
+    val now = clock.instant()
+    val userId = currentUser().userId
+
+    val row =
+        CohortsRow(
+            createdBy = userId,
+            createdTime = now,
+            modifiedBy = userId,
+            modifiedTime = now,
+            name = model.name,
+            phaseId = model.phase)
+
+    cohortsDao.insert(row)
+
+    return row.toModel()
+  }
+
+  fun delete(cohortId: CohortId) {
+    requirePermissions { deleteCohort(cohortId) }
+
+    try {
+      dslContext.transaction { _ ->
+        val rowsDeleted = dslContext.deleteFrom(COHORTS).where(COHORTS.ID.eq(cohortId)).execute()
```

Comment:
This is 100% fine, but FYI in case you didn't know it existed, the DAO classes have a `deleteById` method.

Reply:
I copied this from `ParticipantStore`. It seems like this route was chosen because the DAO `deleteById` method does not return the rows deleted, so we wouldn't be able to throw `CohortNotFoundException` if a cohort that doesn't exist was requested to be deleted. I am going to leave it as is, but if you think I should use the DAO let me know.

Reply:
Ah, good point. Then this is definitely the right way to do it.